### PR TITLE
misc: init typos and github dependabot for CI workflows

### PR DESCRIPTION
**Pull request by phip1611**
- init <https://github.com/crate-ci/typos> and run it in CI
- add GitHub dependabot for CI workflows

### CODE REVIEW

File: .typos.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+extend-exclude = [
+  # Third party.
+  "build-aux/freestanding-toolchain",
+  "freestanding-headers",
+  "decompressor/cc-runtime",
+  "limine-efi",
+  "tinf",
+  "common/flanterm",
+  "stb",
+]
```
Comment (Mintsuki) [inline .typos.toml:6]: Is this necessary at all considering the workflow doesn't run `./bootstrap`?

File: common/protos/linux.c [hunk 2]
```diff
@@ -226,7 +226,7 @@ struct edd_device_params {
         } __attribute__ ((packed)) xprs;
         struct {
             uint64_t reserved;
-        } __attribute__ ((packed)) htpt;
+        } __attribute__ ((packed)) http;
```
Comment (Mintsuki) [inline common/protos/linux.c]: This is an erroneous correction.

File: flake.nix [hunk 3]
```diff
@@ -46,6 +47,10 @@
 
                 # Nix
                 nixpkgs-fmt
+
+                # Misc
+                # typos is not yet frequently updated in the stable channel
+                nixpkgs-unstable.legacyPackages.${pkgs.system}.typos
```
Comment (Mintsuki) [inline flake.nix:58]: What is this for, exactly?

File: .typos.toml (hunk 1 above)
Comment (phip1611) [reply .typos.toml:6]: Yes. Imagine you run `$ typos` locally, after you ran `./bootstrap` beforehand.

File: common/protos/linux.c (hunk 2 above)
Comment (phip1611) [reply common/protos/linux.c]: good catch, fixed.

File: flake.nix (hunk 3 above)
Comment (phip1611) [reply flake.nix:58]: I'm not exactly sure what you are referring to. These are the dependencies for the Nix shell. It serves you with the environment you need for the project without needing global system state. 

image

File: .typos.toml (hunk 1 above)
Comment (Mintsuki) [reply .typos.toml:6]: Well, in that case shouldn't this include all files/directories listed by `git status --ignored` as being ignored after running `./bootstrap` (and potentially after `./configure --enable-all && make` as well)?

File: flake.nix (hunk 3 above)
Comment (Mintsuki) [reply flake.nix:58]: Makes sense.

File: .typos.toml (hunk 1 above)
Comment (phip1611) [reply .typos.toml:6]: I updated the PR. We can remove the excludes as the are coming from `.gitignore` and typos automatically uses common ignore files

**Comment by Mintsuki** [comment]
LGTM thanks!

**Pull request merged**